src/Featurizers/LagLeadOperatorFeaturizer.h

```diff
@@ -0,0 +1,224 @@
+        }
+}
+
+template <typename T>
+LagLeadOperatorTransformer<T>::LagLeadOperatorTransformer(Archive &ar) :
+    LagLeadOperatorTransformer(
+        [&ar](void) {
+            // Version
+            std::uint16_t                   majorVersion(Traits<std::uint16_t>::deserialize(ar));
+            std::uint16_t                   minorVersion(Traits<std::uint16_t>::deserialize(ar));
+
+            if(majorVersion != 1 || minorVersion != 0)
+                throw std::runtime_error("Unsupported archive version");
+
+            // Data
+            std::uint32_t horizon = Traits<std::uint32_t>::deserialize(ar);
+            std::vector<std::int64_t> offsets = Traits<std::vector<std::int64_t>>::deserialize(ar);
+            
+            return LagLeadOperatorTransformer(std::move(horizon), std::move(offsets));
+        }()
+    ) {
+}
+
+template <typename T>
+bool LagLeadOperatorTransformer<T>::operator==(LagLeadOperatorTransformer const &other) const {
+    return this->_horizon == other._horizon
+            && this->_offsets == other._offsets;
+}
+
+template <typename T>
+bool LagLeadOperatorTransformer<T>::operator!=(LagLeadOperatorTransformer const &other) const {
+    return (*this == other) == false;
+}
+
+template <typename T>
+void LagLeadOperatorTransformer<T>::save(Archive &ar) const /*override*/ {
+    // Version
+    Traits<std::uint16_t>::serialize(ar, 1); // Major
+    Traits<std::uint16_t>::serialize(ar, 0); // Minor
+
+    // Data
+    Traits<std::uint32_t>::serialize(ar, _horizon);
+    Traits<std::vector<std::int64_t>>::serialize(ar, _offsets);
+
+    // Note that we aren't serializing working state
+}
+
+template <typename T>
+void LagLeadOperatorTransformer<T>::execute_helper(nonstd::optional<typename BaseType::InputType> const &input, typename BaseType::CallbackFunction const &callback) {
+    _buffer.push(input);
+    // until the circular buffer is full, we don't have enough data to generate output matrix
+    if (_buffer.is_full()) {
+        Microsoft::Featurizer::RowMajMatrix<nonstd::optional<T>> ret(_offsets.size(), _horizon);
+        for (std::uint32_t row = 0; row < _offsets.size(); ++row) {
+            // the circular buffer starts with the needed data for the _max_prior_offset
+            // so the difference between offsets[row] and _max_prior_offset is the position to find the data to copy
+            std::tuple<typename Components::CircularBuffer<TargetType>::iterator, typename Components::CircularBuffer<TargetType>::iterator> range = _buffer.range(_horizon, static_cast<size_t>(_offsets[row] - _max_prior_offset));
+            typename Components::CircularBuffer<TargetType>::iterator start_iter = std::get<0>(range);
+            for (std::int32_t col = 0; col < static_cast<std::int32_t>(_horizon); ++col) {
+                ret(static_cast<std::int32_t>(row), col) = *start_iter;
+                ++start_iter;
+            }
+        }
+        callback(ret);
+    }
```

Comment:
Nit: Add vertical whitespace to help provide context. At the very least, add white space between variable declarations and code.